pkg/globalconfig/global_config.go

```diff
@@ -75,7 +76,14 @@ type GlobalConfig struct {
 	RouterMailpitHTTPSPort           string                      `yaml:"mailpit_https_port,omitempty"`
 	Messages                         MessagesConfig              `yaml:"messages,omitempty"`
 	RemoteConfig                     RemoteConfig                `yaml:"remote_config,omitempty"`
-	ProjectList                      map[string]*ProjectInfo     `yaml:"project_info"`
+}
+
+// Old versions of DDEV stored the project list in the global config.
+// For ease of upgrade, we need to be able to extract that data and move it to
+// the new project_list.yaml
+// In a future release, remove this.
+type LegacyGlobalConfig struct {
+	ProjectList map[string]*ProjectInfo `yaml:"project_info"`
```

Comment:
As above, tempted to allow the ProjectList in current global config, just not write it (just make sure it's empty and `omitempty`